src/super_gradients/training/utils/callbacks/callbacks.py

```diff
@@ -1112,6 +1112,162 @@ def _is_more_extreme(self, score: float) -> bool:
+        max: bool = False,
+        freq: int = 1,
+        classes: Optional[List[str]] = None,
+        normalize_targets: bool = False,
+    ):
+        super(ExtremeBatchDetectionVisualizationCallback, self).__init__(
+            metric=metric, metric_component_name=metric_component_name, loss_to_monitor=loss_to_monitor, max=max, freq=freq
+        )
+        self.post_prediction_callback = post_prediction_callback
+        self.classes = classes
+        self.normalize_targets = normalize_targets
+
+    @staticmethod
+    def universal_undo_preprocessing_fn(inputs):
+        """
+        A universal reversing of preprocessing to be passed to DetectionVisualization.visualize_batch's undo_preprocessing_func kwarg.
+        :param inputs:
+        :return:
+        """
+        inputs -= inputs.min()
+        inputs /= inputs.max()
+        inputs *= 255
+        inputs = inputs.to(torch.uint8)
+        inputs = inputs.cpu().numpy()
+        inputs = inputs[:, ::-1, :, :].transpose(0, 2, 3, 1)
+        inputs = np.ascontiguousarray(inputs, dtype=np.uint8)
+        return inputs
+
+    def process_extreme_batch(self) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Processes the extreme batch, and returns 2 image batches for visualization - one with predictions and one with GT boxes.
+        :return:Tuple[np.ndarray, np.ndarray], the predictions batch, the GT batch
+        """
+        inputs = self.extreme_batch
+        preds = self.post_prediction_callback(self.extreme_preds, self.extreme_batch.device)
+        targets = self.extreme_targets.clone()
+        if self.normalize_targets:
+            target_bboxes = targets[:, 2:]
+            target_bboxes = cxcywh2xyxy(target_bboxes)
+            _, _, height, width = inputs.shape
+            target_bboxes[:, [0, 2]] /= width
+            target_bboxes[:, [1, 3]] /= height
+            target_bboxes = xyxy2cxcywh(target_bboxes)
+            targets[:, 2:] = target_bboxes
+
+        images_to_save_preds = DetectionVisualization.visualize_batch(
+            inputs, preds, targets, "extreme_batch_preds", self.classes, gt_alpha=0.0, undo_preprocessing_func=self.universal_undo_preprocessing_fn
+        )
+        images_to_save_preds = np.stack(images_to_save_preds)
+
+        images_to_save_gt = DetectionVisualization.visualize_batch(
+            inputs, None, targets, "extreme_batch_gt", self.classes, gt_alpha=1.0, undo_preprocessing_func=self.universal_undo_preprocessing_fn
+        )
+        images_to_save_gt = np.stack(images_to_save_gt)
+
+        return images_to_save_preds, images_to_save_gt
+
+    def on_validation_loader_end(self, context: PhaseContext) -> None:
+        if self.classes is None:
+            if hasattr(context.valid_loader.dataset, "classes"):
+                self.classes = context.valid_loader.dataset.classes
+
+            else:
+                raise RuntimeError("Couldn't fetch classes from valid_loader, please pass classes explicitly")
```

Comment:
This is just a fallback in case no classes have been passed on init.
This is state in the docs, but in case it was not clear I added a message saying the calsses attribute will be fetched from validation when none is passed.